scriptworker/test/test_utils.py

```diff
@@ -312,6 +312,53 @@ def test_rm_dir():
     assert not os.path.exists(tmp)
 
 
+# write_to_file {{{1
+@pytest.mark.parametrize('file_type,contents,expected', ((
+    'json', {'a': 'b'}, """{
+  "a": "b"
+}"""
+), (
+    'binary', b'asdfasdf', b'asdfasdf'
+), (
+    'text', 'asdfasdf', 'asdfasdf'
+)))
+def test_write_to_file(tmpdir, file_type, contents, expected):
+    path = os.path.join(tmpdir, 'foo')
+    utils.write_to_file(path, contents, file_type=file_type)
+    mode = 'rb' if file_type == 'binary' else 'r'
```

Comment:
I think this should always be binary, to test exactly what got written. And maybe there should be some non-ascii data, and non-UTF-8 binary data.

Reply:
> I think this should always be binary, to test exactly what got written.

How do we verify that it was written as non-binary, then?

Reply:
I've added additional parameters to test, but haven't yet changed the mode.

Reply:
> > I think this should always be binary, to test exactly what got written.
> 
> How do we verify that it was written as non-binary, then?

I think there is an incorrect assumption behind this question. In every case, things end up as bytes on disk. Opening in binary or not changes how things get converted from the internal representation to bytes on disk; opening as text means you can write `str` but not `bytes` and opening as binary vice-versa.

Reply:
Done.